cluster/gce/config-default.sh

```diff
@@ -115,6 +115,9 @@ DNS_SERVER_IP="${KUBE_DNS_SERVER_IP:-10.0.0.10}"
 DNS_DOMAIN="${KUBE_DNS_DOMAIN:-cluster.local}"
 DNS_REPLICAS=1
 
+# Optional: Enable DNS horizontal autoscaler
+ENABLE_DNS_HORIZONTAL_AUTOSCALER="${KUBE_ENABLE_DNS_HORIZONTAL_AUTOSCALER:-true}"
```

Comment:
you also need to update the gke configs: https://github.com/kubernetes/kubernetes/blob/master/cluster/gke/config-common.sh#L39 (follow up pr is fine if you prefer but I guess you need to specify limits in this one anyway). 
Do all other cloudproviders handle this gracefully? or do you have to pipe some env var down into them as well?